src/components/music/SongPianoRoll.tsx

```diff
@@ -345,22 +1021,39 @@ export const SongPianoRoll = ({
                   </>
                 ))}
             </Piano>
-            <SongGrid>
+            <SongGrid
+              onMouseDown={(e) => {
+                handleMouseDown(e.nativeEvent);
+              }}
+              onMouseLeave={(e) => {
+                handleMouseLeave(e.nativeEvent);
+              }}
+              // onMouseMove={(e) => {
```

Comment:
This needs to be consolidated. As each component was handling mouse move at a different scope (global vs component) right now dragging notes or moving the mouse out of bounds leads to some weird behaviors.